src/main/java/de/tum/in/www1/artemis/service/plagiarism/PlagiarismCaseService.java

```diff
@@ -0,0 +1,158 @@
+     * @return                      the plagiarism case with the verdict
+     */
+    public PlagiarismCase updatePlagiarismCaseVerdict(long plagiarismCaseId, PlagiarismVerdictDTO plagiarismVerdictDTO) {
+        Optional<PlagiarismCase> optionalPlagiarismCase = plagiarismCaseRepository.findById(plagiarismCaseId);
+        if (optionalPlagiarismCase.isEmpty()) {
+            throw new EntityNotFoundException("Plagiarism Case", plagiarismCaseId);
+        }
+        PlagiarismCase plagiarismCase = optionalPlagiarismCase.get();
+        plagiarismCase.setVerdict(plagiarismVerdictDTO.getVerdict());
+        if (plagiarismVerdictDTO.getVerdict().equals(PlagiarismVerdict.POINT_DEDUCTION)) {
+            plagiarismCase.setVerdictPointDeduction(plagiarismVerdictDTO.getVerdictPointDeduction());
+        }
+        else if (plagiarismVerdictDTO.getVerdict().equals(PlagiarismVerdict.WARNING)) {
+            plagiarismCase.setVerdictMessage(plagiarismVerdictDTO.getVerdictMessage());
+        }
+        plagiarismCase.setVerdictDate(ZonedDateTime.now());
+        var user = userRepository.getUserWithGroupsAndAuthorities();
+        plagiarismCase.setVerdictBy(user);
+        plagiarismCaseRepository.save(plagiarismCase);
+        // Notify the student about the verdict
+        singleUserNotificationService.notifyUserAboutPlagiarismCaseVerdict(plagiarismCase, plagiarismCase.getStudent());
+        return plagiarismCase;
+    }
+
+    /**
+     * Save a post for a plagiarism case and notify the student about the plagiarism case.
+     *
+     * @param plagiarismCaseId  the ID of the plagiarism case for which to save the post
+     * @param post              the post which belongs to the plagiarism case
+     */
+    public void savePostForPlagiarismCaseAndNotifyStudent(long plagiarismCaseId, Post post) {
+        PlagiarismCase plagiarismCase = plagiarismCaseRepository.findByIdWithExerciseAndPlagiarismSubmissionsElseThrow(plagiarismCaseId);
+        plagiarismCase.setPost(post);
+        plagiarismCaseRepository.save(plagiarismCase);
+        singleUserNotificationService.notifyUserAboutNewPlagiarismCase(plagiarismCase, plagiarismCase.getStudent());
+    }
+
+    /**
+     * Create or add a plagiarism case for both students involved in a plagiarism comparison if it is determined to be plagiarism.
+     * For each student the following logic applies:
+     * <ul>
+     *     <li>Create a new plagiarism case if the student isn't already part of a plagiarism case in the exercise</li>
+     *     <li>Add the submission of the student to existing plagiarism case otherwise</li>
+     * </ul>
+     *
+     * @param plagiarismComparisonId    the ID of the plagiarism comparison
+     */
+    public void createOrAddPlagiarismCasesForComparison(long plagiarismComparisonId) {
+        var plagiarismComparison = plagiarismComparisonRepository.findByIdWithSubmissionsStudentsElseThrow(plagiarismComparisonId);
+        // handle student A
```

Comment:
Couldn't this logic be extracted into a method that is then called twice? To me, the student A and B blocks look pretty much identical.

Reply:
I thought about that, too and experimented a bit but there is not meaningful to do it as it operates on two different properties (submissionA and submissionB). But let me know if you're able to think of something 😃

Reply:
Couldn't you just pass the relevant submission and the comparison as parameters to such a method? As far as I can see you only really rely on those objects in these code blocks